### PR TITLE
Add required logging of SRC-20 to SRC-6 examples

### PR DESCRIPTION
**Pull request by SwayStar123**
## Type of change

- Bug fix

## Changes

The following changes have been made:

- Adds logging of values of SRC-20 whenever they are updated in SRC-6 examples
- Formats the examples of src6
- Fixes some inline comments

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
   - [ ] If my change requires substantial documentation changes, I have [requested support from the DevRel team](https://github.com/FuelLabs/devrel-requests/issues/new/choose)
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.
- [x] I have updated the changelog to reflect the changes on this PR.

### CODE REVIEW

File: examples/src6-vault/single_asset_single_sub_vault/src/main.sw [hunk 1]
```diff
@@ -225,4 +278,13 @@ pub fn _burn(asset_id: AssetId, amount: u64) {
     let supply = storage.total_supply.read();
     storage.total_supply.write(supply - amount);
     burn(PRE_CALCULATED_SHARE_VAULT_SUB_ID, amount);
+    TotalSupplyEvent::new(
+        vault_assetid(),
+        storage
```
Comment (bitzoic) [inline examples/src6-vault/single_asset_single_sub_vault/src/main.sw:283]: You already read this from storage on line 279

File: examples/src6-vault/single_asset_single_sub_vault/src/main.sw [hunk 2]
```diff
@@ -211,6 +255,15 @@ pub fn _mint(recipient: Identity, amount: u64) {
     let supply = storage.total_supply.read();
     storage.total_supply.write(supply + amount);
     mint_to(recipient, PRE_CALCULATED_SHARE_VAULT_SUB_ID, amount);
+    TotalSupplyEvent::new(
+        vault_assetid(),
+        storage
```
Comment (bitzoic) [inline examples/src6-vault/single_asset_single_sub_vault/src/main.sw:260]: You already read this from storage on line 255

File: examples/src6-vault/multi_asset_vault/src/main.sw [hunk 3]
```diff
@@ -247,4 +301,14 @@ pub fn _burn(asset_id: AssetId, vault_sub_id: SubId, amount: u64) {
     let supply = storage.total_supply.get(asset_id).try_read().unwrap();
     storage.total_supply.insert(asset_id, supply - amount);
     burn(vault_sub_id, amount);
+    TotalSupplyEvent::new(
+        asset_id,
+        storage
```
Comment (bitzoic) [inline examples/src6-vault/multi_asset_vault/src/main.sw:306]: You already have some of this from line 301, no need to make another expensive storage call

File: examples/src6-vault/multi_asset_vault/src/main.sw [hunk 4]
```diff
@@ -233,6 +277,16 @@ pub fn _mint(
         .total_supply
         .insert(asset_id, supply.unwrap_or(0) + amount);
     mint_to(recipient, vault_sub_id, amount);
+    TotalSupplyEvent::new(
+        asset_id,
+        storage
```
Comment (bitzoic) [inline examples/src6-vault/multi_asset_vault/src/main.sw:282]: This is already fetched

File: examples/src6-vault/single_asset_single_sub_vault/src/main.sw [hunk 5]
```diff
@@ -178,6 +191,37 @@ impl SRC20 for Contract {
...
+    fn set_src20_data(
+        asset: AssetId,
+        name: Option<String>,
+        symbol: Option<String>,
+        decimals: u8,
+    );
+}
+
+impl SetSRC20Data for Contract {
+    #[storage(read)]
```
Comment (bitzoic) [inline examples/src6-vault/single_asset_single_sub_vault/src/main.sw:205]: This is labeled "set" but only reads from storage. Where is the contract setting this information?

File: examples/src6-vault/multi_asset_vault/src/main.sw [hunk 6]
```diff
@@ -170,6 +183,37 @@ impl SRC20 for Contract {
...
+        asset: AssetId,
+        name: Option<String>,
+        symbol: Option<String>,
+        decimals: u8,
+    );
+}
+
+impl SetSRC20Data for Contract {
+    #[storage(read)]
+    fn set_src20_data(
```
Comment (bitzoic) [inline examples/src6-vault/multi_asset_vault/src/main.sw:198]: This function is incomplete

File: examples/src6-vault/single_asset_vault/src/main.sw [hunk 7]
```diff
@@ -184,6 +197,37 @@ impl SRC20 for Contract {
...
+        asset: AssetId,
+        name: Option<String>,
+        symbol: Option<String>,
+        decimals: u8,
+    );
+}
+
+impl SetSRC20Data for Contract {
+    #[storage(read)]
+    fn set_src20_data(
```
Comment (bitzoic) [inline examples/src6-vault/single_asset_vault/src/main.sw:212]: Again, incomplete

File: examples/src6-vault/single_asset_vault/src/main.sw [hunk 8]
```diff
@@ -248,6 +292,16 @@ pub fn _mint(
         .total_supply
         .insert(asset_id, current_supply + amount);
     mint_to(recipient, vault_sub_id, amount);
+    TotalSupplyEvent::new(
+        asset_id,
+        storage
```
Comment (bitzoic) [inline examples/src6-vault/single_asset_vault/src/main.sw:297]: Already read

File: examples/src6-vault/single_asset_vault/src/main.sw [hunk 9]
```diff
@@ -262,4 +316,14 @@ pub fn _burn(asset_id: AssetId, vault_sub_id: SubId, amount: u64) {
     let supply = storage.total_supply.get(asset_id).try_read().unwrap();
     storage.total_supply.insert(asset_id, supply - amount);
     burn(vault_sub_id, amount);
+    TotalSupplyEvent::new(
+        asset_id,
+        storage
```
Comment (bitzoic) [inline examples/src6-vault/single_asset_vault/src/main.sw:321]: Unnecessary read

**Review by bitzoic — CHANGES_REQUESTED** [review]

File: examples/src6-vault/single_asset_single_sub_vault/src/main.sw (hunk 5 above)
Comment (SwayStar123) [reply examples/src6-vault/single_asset_single_sub_vault/src/main.sw:205]: Right, i copy pasted this from the src20 examples, which are the same, those should be fixed too then

**Review by K1-R1 — APPROVED** [review]

**Review by bitzoic — APPROVED** [review]

**Pull request merged**